cmd/podman/images/utils_linux.go

```diff
@@ -0,0 +1,48 @@
+package images
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+var errWaitCancel = errors.New("wait canceled")
+
+// setupPipe() for fixing https://github.com/containers/podman/issues/7017
+// uses named pipe since containers/image EvalSymlinks fails with /dev/stdout
+func setupPipe() (string, func() <-chan error, error) {
```

Comment:
Non-blocking: The semantics of the cleanup callback is now complex enough that it might be worth documenting. (OTOH it’s not _that_ much code and it has a single user.)